solarforecastarbiter/metrics/tests/test_preprocessing.py

```diff
@@ -1653,3 +1670,185 @@ def test_apply_fill_unstratified_dataframe(data, method, exp, exp_count,
                                              data.index[0], data.index[-1])
     pd.testing.assert_frame_equal(result, exp)
     assert count == exp_count
+
+
+OUTAGE_FORECAST = datamodel.Forecast(
+   site=_site_metadata(), name='dummy fx', variable='ghi',
+   interval_value_type='instantaneous',
+   interval_length=pd.Timedelta('1h'),
+   interval_label='beginning',
```

Comment:
my brain is fried at the moment so I will ask the question instead of thinking through it: do we need to account for interval label of forecasts or observations when identifying the times to discard? No? Only a function of the issue time and the lead time?

Reply:
Good catch. I believe we do.
For instance, a forecast with 1hr interval, run length and lead time forecast an issue time at 12:00 would mean dropping 13:00 for a beginning labeled forecast and 14:00 for a ending labelled forecast.